Networking Demo AppTests/APIClientTests.swift

```diff
@@ -145,6 +145,36 @@ class APIClientTests: XCTestCase {
         await waitForExpectations(timeout: 10, handler: nil)
     }
     
+    @available(iOS 15.0.0, *)
+    func testAsyncCallbackFailedDownloadRequest() async throws {
+        var request = TestAsyncDownloadRequest { progress in
+            XCTFail("Callback should not be triggered for failed download.")
+        }
+        request.path = "kljhfkajsdhfs/////df345.345345m34feg45435"
+        let api = APIClient(apiContext: SimpleAPIContext())
+        
+        do {
+            let _ = try await api.perform(request)
+            XCTFail("Error was not thrown as it should be.")
+        } catch let error {
+            XCTAssertNotNil(error)
```

Comment:
can error be `nil` in this case? maybe we can do some assertions on the error itself?
same for line 174

Reply:
Good catch, indeed that could be a better assert.